src/scripts/us-code.js

```diff
@@ -0,0 +1,192 @@
+    content: dom("> .content, > .chapeau", node).text().trim() || false,
+    children: [
+      ...dom("> .paragraph, > .subparagraph, > .clause, > .subclause", node),
+    ].map((c) => parseSubcomponent(c, dom)),
+  };
+
+  // If the subcomponent has a heading, tack that on to the component name
+  const heading = dom("> .heading", node).text().trim();
+  if (heading.length > 0) {
+    subsection.name = `${subsection.name} ${heading}`;
+  }
+
+  return subsection;
+};
+
+const stringifySubcomponent = (subcomponent, depth) => {
+  // Slack collapses multiple spaces into one, so we'll use the :blank: emoji
+  // to get indentation.
+  const indent = [...Array(depth)].map(() => ":blank:").join("");
+
+  const text = [];
+  if (subcomponent.content) {
+    text.push(`${indent}*${subcomponent.name}* ${subcomponent.content}`);
+  } else {
+    text.push(`${indent}*${subcomponent.name}*`);
+  }
+
+  for (const subComponent of subcomponent.children) {
+    text.push(stringifySubcomponent(subComponent, depth + 1));
+  }
+
+  return text.join("\n");
+};
+
+module.exports = (app) => {
+  registerInteractive(
+    "US Code bot",
+    "[title] USC [section] (<maybe>)(<paragraphs>)(<too>)",
+    "If you drop a citation to the US Code, Charlie will attempt to look it up on the <https://www.law.cornell.edu/uscode/text|Cornell LII> website and post the text in a thread. Handy!",
+    false
+  );
+
+  app.message(
+    trigger,
+    async ({
+      context: { matches },
+      event: { thread_ts: thread, ts },
+      message: { text: message },
+      say,
+    }) => {
+      const titleNumber = +matches[1];
+      const sectionNumber = +matches[2];
+
+      const subcitations = (message.match(
+        // This matcher pulls out not just the title & section, but also any
+        // deeper references (paragraphs, subparagraphs, etc.), as long as they
+        // are in parantheses after the section. We're only going to keep the
+        // first match, which is the entire citation.
+        // https://regexper.com/#%2F%28%5Cd%2B%29%5Cs*u%5C.%3Fs%5C.%3Fc%5C.%3F%5Cs*%C2%A7%3F%5Cs*%28%5Cd%2B%29%28%5Cs*%5C%28%28%5Ba-z%5D%7Ci%2B%7C%5Cd%2B%29%5C%29%29*%2Fi
+        /(\d+)\s*u\.?s\.?c\.?\s*§?\s*(\d+)(\s*\(([a-z]|i+|\d+)\))*/i
+      ) ?? [""])[0]
+        // Now remove the [title] usc [section] part, because we've already
+        // parsed that part out.
+        .replace(trigger, "")
+        // Our regex allows arbitrary whitespace between the citation components
+        // but those make it harder to parse, so we'll just get rid of it all
+        .replace(/\s/g, "")
+        // Get rid of the opening paren and split on the closing paren to get
+        // all the components into an array.
+        .replace(/\(/g, "")
+        .split(")")
+        // Get rid of the last element because it will always be empty since the
+        // regex requires that citation components end with a closing paren.
+        .slice(0, -1)
+        .map((v) => v.toLowerCase());
```

Comment:
i wish that javascript allowed you to return multiple match groups with `.match`. 

However, it may still be possible to simplify this somewhat, if you think it might be more readable.

```
message.replace(trigger, "")
  \\ this match will return an array of the paragraphs still in parens, ie ['(b)', '(3)', '(F)', '(iii)', '(I)']
  .match(/\((\w*)\)+/g)
  .map((v) =>v.replace("(", "").replace(")", "")).toLowerCase())
``` 
I'm not 100% convinced this is better, but figured I'd throw it out there and see if ya like it.

Reply:
I had written a response to this and then GitHub ate it...

Strong agree that the current code is a bit of a mess. It turns out that the complication is because the sub-citations (the parentheticals) need to be anchored to the main citation. This suggestion is definitely easier to understand, but without being anchored, this text is troublesome:

```
42 USC 3100 (a)(b) is what he (Fred) likes to read for fun
```

It ends up thinking that `(Fred)` is the last component of the citation and would result in some funky output. I added a test to catch Fred's 🤓 nerdery, and also added more comments about the anchoring to hopefully help someone in the future know why this is what it is.